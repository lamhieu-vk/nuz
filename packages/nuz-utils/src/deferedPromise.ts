export interface DeferedPromise<T extends unknown> extends Promise<T> {
  promise: Promise<T>
  resolve: (value: T) => void
  reject: (error: Error) => void
}

function deferedPromise<T extends unknown>(): DeferedPromise<T> {
  const defered = {} as DeferedPromise<T>

  defered.promise = new Promise((resolve, reject) => {
    defered.resolve = resolve
    defered.reject = reject
  })

  return defered
}

export default deferedPromise
