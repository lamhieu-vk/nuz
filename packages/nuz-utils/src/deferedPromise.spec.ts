import * as func from './deferedPromise'
import deferedPromise from './deferedPromise'

describe('deferedPromise', () => {
  it('Exported as default', () => {
    expect(func.default).toBeDefined()
  })

  it('Should match to snapshot', () => {
    expect(deferedPromise()).toMatchSnapshot()
  })
})
