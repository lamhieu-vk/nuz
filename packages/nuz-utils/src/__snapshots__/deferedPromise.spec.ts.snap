// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deferedPromise Should match to snapshot 1`] = `
Object {
  "promise": Promise {},
  "reject": [Function],
  "resolve": [Function],
}
`;
